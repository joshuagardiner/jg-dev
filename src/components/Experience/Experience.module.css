.cardColumnCollection {
  display: inline-flex;
}

.company {
  font-size: 1rem;
  text-transform: uppercase;
}

.content {
  display: inline-block;
  font-size: 1.125rem;
  line-height: 2.75rem;
  position: relative;
  width: 90%;
}

.divider {
  display: none;
}

.role {
  font-size: 1.125rem;
  text-transform: uppercase;
}

.subExperience {
  width: 50%;
}

.tenure {
  font-size: 0.75rem;
  text-transform: uppercase;
}

.title {
  font-size: 2rem;
  margin-top: 5%;
}

@media screen and (max-width: 768px) {
  .cardColumnCollection {
    display: grid;
  }

  .divider {
    color: #3c4858;
    display: block;
    width: 5%;
  }

  .subExperience {
    width: 100%;
  }
}

@media screen and (max-width: 425px) {
  .company {
    font-size: 0.66rem;
  }

  .content {
    font-size: 0.75rem;
    line-height: 1.5rem;
    width: 95%;
  }

  .role {
    font-size: 0.75rem;
  }

  .tenure {
    font-size: 0.5rem;
  }

  .title {
    font-size: 1.25rem;
  }
}
