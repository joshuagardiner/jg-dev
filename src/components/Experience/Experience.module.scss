.company {
  font-size: 1rem;
  text-transform: uppercase;
}

.content {
  display: inline-block;
  font-size: 1.125rem;
  line-height: 2.75rem;
  width: 80%;
}

.divider {
  display: none;
}

.role {
  font-size: 1.125rem;
  text-transform: uppercase;
}

.tenure {
  font-size: 0.75rem;
  text-transform: uppercase;
}

@media screen and (max-width: 768px) {
  .content {
    width: 100%;
  }

  .divider {
    color: #3c4858;
    display: block;
    width: 5%;
  }
}

@media screen and (max-width: 425px) {
  .company {
    font-size: 0.66rem;
  }

  .content {
    font-size: 0.75rem;
    line-height: 1.5rem;
  }

  .role {
    font-size: 0.75rem;
  }

  .tenure {
    font-size: 0.5rem;
  }
}
