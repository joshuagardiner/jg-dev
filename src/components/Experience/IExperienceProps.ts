import { IExperienceContent } from "../../lib/models/IExperience";

/**
 * The properties available for the Experience component.
 */
export interface IExperienceProps {
  /**
   * The company of employment.
   */
  company: string;

  /**
   * The experiences gained.
   */
  content: IExperienceContent[];

  /**
   * The role of employment.
   */
  role: string;

  /**
   * The timeframe of employment.
   */
  tenure: string;
}
