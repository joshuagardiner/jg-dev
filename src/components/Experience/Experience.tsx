import React from "react";
import { Card, Collection, CollectionColumn } from "@joshuagardiner/typescript-component-library";
import { IExperienceProps } from "./IExperienceProps";
import styling from "./Experience.module.css";

/**
 * The Experience component is responsible for rendering content
 * within the application Body.
 *
 * @returns the Experience component.
 */
export const Experience: React.FC<IExperienceProps> = (props: IExperienceProps) => {
  const { experiences } = props;

  return experiences ? (
    <div data-testid={"experience-component"}>
      <h2 className={styling.title} data-testid={"experience-title"}>
        Experience
      </h2>
      <Collection>
        <Card>
          <h3 className={styling.role} data-testid={"experience-role"}>
            {experiences[0].role}
          </h3>
          <h4 className={styling.company} data-testid={"experience-company"}>
            {experiences[0].company}
          </h4>
          <h5 className={styling.tenure} data-testid={"experience-tenure"}>
            {experiences[0].tenure}
          </h5>
          <p className={styling.content} data-testid={"experience-content"}>
            {experiences[0].content[0]}
            <br></br>
            <br></br>
            {experiences[0].content[1]}
            <br></br>
            <br></br>
            {experiences[0].content[2]}
            <br></br>
            <br></br>
            {experiences[0].content[3]}
            <br></br>
            <br></br>
            {experiences[0].content[4]}
            <br></br>
            <br></br>
            {experiences[0].content[5]}
          </p>
        </Card>
        <CollectionColumn className={styling.cardColumnCollection}>
          <Card className={styling.subExperience}>
            <h3 className={styling.role} data-testid={"experience-role"}>
              {experiences[1].role}
            </h3>
            <h4 className={styling.company} data-testid={"experience-company"}>
              {experiences[1].company}
            </h4>
            <h4 className={styling.tenure} data-testid={"experience-tenure"}>
              {experiences[1].tenure}
            </h4>
            <p className={styling.content} data-testid={"experience-sub-content"}>
              {experiences[1].content[0]}
            </p>
          </Card>
          <Card className={styling.subExperience}>
            <h3 className={styling.role} data-testid={"experience-role"}>
              {experiences[2].role}
            </h3>
            <h4 className={styling.company} data-testid={"experience-company"}>
              {experiences[2].company}
            </h4>
            <h4 className={styling.tenure} data-testid={"experience-tenure"}>
              {experiences[2].tenure}
            </h4>
            <p className={styling.content} data-testid={"experience-sub-content"}>
              {experiences[2].content[0]}
            </p>
          </Card>
        </CollectionColumn>
      </Collection>
    </div>
  ) : (
    <></>
  );
};
