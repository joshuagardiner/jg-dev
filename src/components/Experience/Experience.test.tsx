import { render, screen } from "@testing-library/react";
import { Experience } from "./Experience";

describe("Experience", () => {
  it("renders the Experience component correctly.", () => {
    render(<Experience />);

    const experienceTitles = screen.getAllByTestId("title");
    const experienceCards = screen.getAllByTestId("card");
    const experienceText = screen.getAllByTestId("text");

    expect(experienceTitles).toBeDefined();
    expect(experienceTitles).toHaveLength(10);
    expect(experienceTitles[0].textContent).toEqual("Experience");
    expect(experienceTitles[1].textContent).toEqual("Software Development Engineer");
    expect(experienceTitles[2].textContent).toEqual("EMIS Health");
    expect(experienceTitles[3].textContent).toEqual("November '17 - Present");
    expect(experienceTitles[4].textContent).toEqual(".NET Developer");
    expect(experienceTitles[5].textContent).toEqual("FDM Group");
    expect(experienceTitles[6].textContent).toEqual("May '17 - October '17");
    expect(experienceTitles[7].textContent).toEqual("Customer Service Assistant");
    expect(experienceTitles[8].textContent).toEqual("Sainsbury's");
    expect(experienceTitles[9].textContent).toEqual("December '13 - June '15");
    expect(experienceCards).toBeDefined();
    expect(experienceCards).toHaveLength(4);
    expect(experienceText).toBeDefined();
    expect(experienceText).toHaveLength(8);
    expect(experienceText[0].textContent).toEqual(
      "I have enjoyed working with a wide ranging, ever changing tech stack during my time at the company which began in C#, involving the refactoring of a legacy product alongside resolving priority SLA items."
    );
    expect(experienceText[1].textContent).toEqual(
      "I was a member of the first set of engineers to begin coding the next generation of product for the business which allowed a great deal of flexibility and freedom as we identified new tools and services fit for use. Including as a learning exercise the creation of a Twitter bot (@motivationalboz) using Node.js and AWS Lambda to post a tweet on a CRON schedule."
    );
    expect(experienceText[2].textContent).toEqual(
      "After this period of elaboration I began to work exclusively in this space with Golang and the varying services AWS provide, inc. API Gateway, CloudFormation, CodePipeline and Lambda. Writing one of the first CloudFormation templates which deployed a CI/CD pipeline in the department."
    );
    expect(experienceText[3].textContent).toEqual(
      "This has organically led to the consumption of APIâ€™s and a more front-end focus with the use of React and TypeScript, and allowed for more hands-on experience with the likes of Terraform and Docker through repository management and containerised applications."
    );
    expect(experienceText[4].textContent).toEqual(
      "I began to contribute to our new and growing community initiatives before identifying an opportunity and leading on my own with the support of my peers, whilst actively looking for areas of improvement not only within the team but within the department by identifying new tools and best practices."
    );
    expect(experienceText[5].textContent).toEqual(
      "I naturally undertook the mentoring of an apprentice developer and test engineer within my team, which led to a company-wide award in the inaugural EMIS Quarterly Heroes Q1 2021 for living the Supportive value of the business."
    );
    expect(experienceText[6].textContent).toEqual(
      "I joined the award winning training programme and undertook weekly modules such as: SQL, Web Development and Professional Skills. Before a more concentrated .NET period which focused on the SOLID principles, Object-oriented programming and TDD. Culminating in a presentation to my peers, and wider department involving the eCommerce project created during this time."
    );
    expect(experienceText[7].textContent).toEqual(
      "Responsible for delivering a first class customer service in any capacity around the store. My role was to in the first instance operate the checkouts, whilst during busier periods I would become the focal point of the checkouts department. Ensuring all colleagues have everything they need to hand, answering any customer queries a colleague could not."
    );
  });

  it("renders the Experience component unchanged.", () => {
    const { container } = render(<Experience />);
    expect(container).toMatchSnapshot();
  });
});
