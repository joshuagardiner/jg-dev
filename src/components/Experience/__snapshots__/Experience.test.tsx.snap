// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders the Experience component unchanged. 1`] = `
<div>
  <div
    data-testid="experience-component"
  >
    <h2
      class="title"
      data-testid="experience-title"
    >
      Experience
    </h2>
    <div
      class="tcl-card"
      data-testid="card"
      style="height: auto; margin-top: auto; max-width: 100%;"
    >
      <div
        class="tcl-card__content"
        data-testid="card-content"
      >
        <h3
          class="role"
          data-testid="experience-role"
        >
          Software Development Engineer
        </h3>
        <h4
          class="company"
          data-testid="experience-company"
        >
          EMIS Health
        </h4>
        <h5
          class="tenure"
          data-testid="experience-tenure"
        >
          November '17 - Present
        </h5>
        <p
          class="content"
          data-testid="experience-content"
        >
          I have enjoyed working with a wide-ranging, ever-changing tech stack during my time at the company beginning in C#, involving the refactoring of a legacy product alongside resolving priority SLA items.
          <br />
          <br />
          I was a member of the first set of engineers to begin coding the next generation of products for the business. Which allowed a great deal of flexibility and freedom as we identified new tools and services fit for use. Including learning exercises such as creating a Twitter bot (@motivationalboz) using Node.js and AWS Lambda to post a tweet on a CRON schedule.
          <br />
          <br />
          After this period of elaboration, I began to work exclusively in this space with Golang and the varying services AWS provide, inc. API Gateway, CloudFormation, CodePipeline and Lambda. Writing one of the first CloudFormation templates to deploy a CI/CD pipeline in the department.
          <br />
          <br />
          This has organically led to the consumption of APIâ€™s and a more front-end focus using React and TypeScript and allowed for more hands-on experience with Terraform and Docker through repository management and containerised applications.
          <br />
          <br />
          I began to contribute to our new and growing community initiatives before identifying an opportunity and leading on my own with the support of my peers, whilst actively looking for areas of improvement not only within the team but within the department by identifying new tools and best practices.
          <br />
          <br />
          I naturally undertook the mentoring of an apprentice developer and test engineer within my team, which led to a company-wide award in the inaugural EMIS Quarterly Heroes Q1 2021 for living the Supportive value of the business.
        </p>
        <div
          class="tcl-card"
          data-testid="card"
          style="height: auto; margin-top: 2vw; max-width: 100%;"
        >
          <div
            class="tcl-card__content"
            data-testid="card-content"
          >
            <div
              class="tcl-card"
              data-testid="card"
              style="height: auto; margin-top: auto; max-width: 50%;"
            >
              <div
                class="tcl-card__content"
                data-testid="card-content"
              >
                <h3
                  class="role"
                  data-testid="experience-role"
                >
                  .NET Developer
                </h3>
                <h4
                  class="company"
                  data-testid="experience-company"
                >
                  FDM Group
                </h4>
                <h4
                  class="tenure"
                  data-testid="experience-tenure"
                >
                  May '17 - October '17
                </h4>
                <p
                  class="subContent"
                  data-testid="experience-sub-content"
                >
                  I joined the award-winning training programme and undertook weekly modules such as SQL, Web Development and Professional Skills. Before a more concentrated .NET period focusing on the SOLID principles, Object-oriented programming and TDD, which concluded in a presentation to my peers and account managers focusing on the eCommerce project created during this time.
                </p>
              </div>
            </div>
            <div
              class="tcl-card"
              data-testid="card"
              style="height: auto; margin-top: auto; max-width: 50%;"
            >
              <div
                class="tcl-card__content"
                data-testid="card-content"
              >
                <h3
                  class="role"
                  data-testid="experience-role"
                >
                  Customer Service Assistant
                </h3>
                <h4
                  class="company"
                  data-testid="experience-company"
                >
                  Sainsbury's
                </h4>
                <h4
                  class="tenure"
                  data-testid="experience-tenure"
                >
                  December '13 - June '15
                </h4>
                <p
                  class="subContent"
                  data-testid="experience-sub-content"
                >
                  Responsible for delivering first-class customer service in any capacity around the store. My role in the first instance was to operate the checkouts. Whilst during the busier period' I would become the focal point of the checkouts department. Ensuring all colleagues have everything they need to hand, answering any customer queries a colleague could not.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
